dali/test/python/test_nvjpeg_cache.py

```diff
@@ -0,0 +1,75 @@
+# Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from nvidia.dali.pipeline import Pipeline
+import nvidia.dali.ops as ops
+import nvidia.dali.types as types
+import nvidia.dali.tfrecord as tfrec
+from timeit import default_timer as timer
+import numpy as np
+import os
+from numpy.testing import assert_array_equal, assert_allclose
+
+seed = 1549361629
+
+img_root = os.environ["DALI_EXTRA_PATH"]
+image_dir = img_root + "/db/single/jpeg"
+batch_size = 20
+
+def compare(tl1, tl2):
+    tl1_cpu = tl1.as_cpu()
+    tl2_cpu = tl2.as_cpu()
+    #tl2 = tl2.as_cpu()
+    assert(len(tl1_cpu) == len(tl2_cpu))
+    for i in range(0, len(tl1_cpu)):
+        assert_array_equal(tl1_cpu.at(i), tl2_cpu.at(i), "cached and non-cached images differ")
+
+class nvJPEGDecoderPipeline(Pipeline):
+    def __init__(self, batch_size, num_threads, device_id, cache_size):
+        super(nvJPEGDecoderPipeline, self).__init__(batch_size, num_threads, device_id, seed = seed)
+        self.input = ops.FileReader(file_root = image_dir)
+        policy = None
+        if cache_size > 0:
+          policy = "threshold"
+        self.decode = ops.nvJPEGDecoder(device = 'mixed', output_type = types.RGB, cache_debug = False, cache_size = cache_size, cache_type = policy, cache_batch_copy = True)
```

Comment:
you could make `cache_batch_copy` and argument and compare the two variants in another test case

Reply:
It's a separate PR.

Reply:
I could, but it's not essential. The two variants of the ScatterGather are already unit-tested.